Content.Server/Radiation/Systems/RadiationSystem.cs

```diff
@@ -32,8 +37,91 @@ public override void Initialize()
 
         _blockerQuery = GetEntityQuery<RadiationBlockingContainerComponent>();
         _resistanceQuery = GetEntityQuery<RadiationGridResistanceComponent>();
-        _gridQuery = GetEntityQuery<MapGridComponent>();
         _stackQuery = GetEntityQuery<StackComponent>();
+
+        SubscribeLocalEvent<RadiationSourceComponent, ComponentInit>(OnSourceInit);
+        SubscribeLocalEvent<RadiationSourceComponent, ComponentShutdown>(OnSourceShutdown);
+        SubscribeLocalEvent<RadiationSourceComponent, MoveEvent>(OnSourceMove);
+        SubscribeLocalEvent<RadiationSourceComponent, StackCountChangedEvent>(OnSourceStackChanged);
+        
+        SubscribeLocalEvent<RadiationReceiverComponent, ComponentInit>(OnReceiverInit);
+        SubscribeLocalEvent<RadiationReceiverComponent, ComponentShutdown>(OnReceiverShutdown);
+    }
+
+    private void OnSourceInit(EntityUid uid, RadiationSourceComponent component, ComponentInit args)
+    {
+        UpdateSource(uid, component);
+    }
```

Comment:
You'll want to change all of these to
```suggestion
    private void OnSourceInit(Entity<RadiationSourceComponent> entity, ref ComponentInit args)
    {
        UpdateSource(entity);
    }
```

Reply:
✅